packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -649,6 +717,56 @@ class _RenderCupertinoSwitch extends RenderConstrainedBox {
       canvas.drawRRect(borderTrackRRect, borderPaint);
     }
 
+    if (_onOffSwitchLabels) {
+      final double leftLabelOpacity = visualPosition * (1.0 - currentReactionValue);
+      final double rightLabelOpacity = (1.0 - visualPosition) * (1.0 - currentReactionValue);
+
+      late final Offset onLabelOffset;
+      late final double onLabelOpacity;
+      late final Offset offLabelOffset;
+      late final double offLabelOpacity;
+
+      switch (textDirection) {
+        case TextDirection.ltr:
+          onLabelOffset =
+              trackRect.centerLeft.translate(_kOnLabelPaddingHorizontal, 0);
```

Comment:
nit: if the parent RenderObject gives this switch tight constraints then the switch's actual dimensions could be very different from `_kSwitchWidth x _kSwitchHeight`.

Reply:
(this is somewhat of an edge case so feel free to keep it as is and open a new issue for it).